File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    vector<int> row; 
    vector<int> column;
    
    for(int i = 0 ; i < sizes.size() ; i++){
        if (sizes[i][0]>sizes[i][1]){
            row.push_back(sizes[i][0]);
            column.push_back(sizes[i][1]);
        }
        else {
            row.push_back(sizes[i][1]);
            column.push_back(sizes[i][0]);
        }
    }
                           
    sort(row.begin(),row.end());
    sort(column.begin(),column.end());
    
    int tmp_r = row.back();
    int tmp_c = column.back();
    
    answer = tmp_r*tmp_c;
    
    
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    
    for i in range (len(sizes)):
        if sizes[i][0] < sizes[i][1]:
            temp = sizes[i][0]
            sizes[i][0] = sizes[i][1]
            sizes[i][1] = temp

    cur_r = sizes[0][0]
    cur_c = sizes[0][1]
    
    for i in range (len(sizes)-1):
        if cur_r < sizes[i+1][0]:
            cur_r = sizes[i+1][0]
            
    for i in range (len(sizes)-1):
        if cur_c < sizes[i+1][1]:
            cur_c = sizes[i+1][1]
            
    answer = cur_r * cur_c
    return answer